dune_client/api/base.py

```diff
@@ -179,15 +179,22 @@ def _get(
             return response
         return self._handle_response(response)
 
-    def _post(self, route: str, params: Optional[Any] = None) -> Any:
+    def _post(
+        self,
+        route: str,
+        params: Optional[Any] = None,
+        data: Optional[IO[bytes]] = None,
+        headers: Optional[Dict[str, Optional[str]]] = None,
+    ) -> Any:
         """Generic interface for the POST method of a Dune API request"""
         url = self._route_url(route)
         self.logger.debug(f"POST received input url={url}, params={params}")
         response = self.http.post(
             url=url,
             json=params,
-            headers=self.default_headers(),
+            headers=dict(self.default_headers(), **(headers or {})),
             timeout=self.request_timeout,
+            data=data,
```

Comment:
both `data` and `json` should use the body. One of them must take precedence? Either way, it could be wise to add that you are not expected to include both params and data to the body? Probably nitpicking, after all its an internal api.

Reply:
Not sure I follow if you are requesting a change here?

Reply:
Either nothing or a small comment in the docstring. I'm leaning towards nothing.